docs/ios/6x/automatic-instrumentation/hang-detection.md

```diff
@@ -0,0 +1,425 @@
+---
+title: Hang Detection
+description: Automatically detect and monitor main thread hangs in your iOS app with Embrace
+sidebar_position: 7
+---
+
+# Hang Detection
+
+The Embrace SDK automatically monitors your application's main thread for hangs (also known as Application Not Responding or ANR events), providing visibility into UI freezes and unresponsive behavior that can frustrate users.
+
+## What are Hangs?
+
+A hang occurs when your app's main thread (UI thread) is blocked for too long, preventing the app from responding to user interactions. During a hang, your app may appear frozen—buttons don't respond, animations stop, and the UI becomes unresponsive.
+
+Common causes of hangs include:
+
+- Performing heavy computations on the main thread
+- Synchronous network requests
+- Large file I/O operations
+- Inefficient database queries
+- Deadlocks or race conditions
+- Slow third-party SDK initialization
+
+Even hangs as short as 250 milliseconds can be noticeable to users and negatively impact the user experience.
+
+## How Hang Detection Works
+
+Embrace uses two complementary mechanisms to detect and report hangs:
+
+### 1. Real-time Hang Detection (HangWatchdog)
+
+The SDK uses a dedicated high-priority watchdog thread that monitors the main `RunLoop` for blockages. This provides:
+
+- **Immediate Detection**: Hangs are detected in real-time as they occur
+- **Configurable Threshold**: By default, hangs lasting longer than 249 milliseconds are captured (Apple's recommended threshold)
+- **Stack Trace Sampling**: Optionally captures periodic stack traces during the hang to identify the blocking code
+- **Continuous Monitoring**: Tracks hang duration and updates while the hang persists
+
+**Detection Mechanism:**
+
+- A `CFRunLoopObserver` monitors the main RunLoop for activity
+- A separate high-priority watchdog thread runs a timer that checks for blockages
+- When the main thread fails to respond within the threshold, a hang is reported
+- The watchdog continues tracking until the main thread becomes responsive again
+
+### 2. MetricKit Hang Reporting
```

Comment:
this isn't supported yet and I'm not sure if we will support it. @ArielDemarco ?